### ISSUE TITLE
Improvement on handling corrupted files

### ISSUE THREAD

**Issue by Brookluo** [Sbozzolo/kuibit#31]
I encountered an issue when kuibit read a corrupted HDF5 (because the program ends before the cactus finished writing to HDF5). When kuitbit reads a corrupted HDF5 file, the reader fail directly and the exception is thrown by the h5py package. Currently, the error message is:

```
RuntimeError: Can't determine # of objects (addr overflow, addr = 1231693139, size = 328, eoa = 1231364739)
```

I think it didn't provide useful debug information about which file caused this issue. Because there are a lot of files in each directory, users might take some effort to dig for more information about the corrupted file. Hence, it might be beneficial to catch the exception from h5py and tell the user which file in which directory caused that error. 

Could you add an additional exception handling after this line to check the integrity of the file, https://github.com/Sbozzolo/kuibit/blob/master/kuibit/cactus_grid_functions.py#L1410
and throw the error again but with additional information about the file name, its path etc?
That will give users more useful debugging information. Thanks.

**Comment by Sbozzolo** [comment]
Implemented in 27c2aff46d5c46faff153219b95ca863dddead42.

**Comment by Brookluo** [comment]
I am having an issue with the current implementation of handling the corrupted data. The file object, rather than the filename string, is currently fed to the error message f-string. This causes problems if the file has been closed. See :
https://github.com/Sbozzolo/kuibit/blob/267005ea1d2788f40196c50498b30a1d231b875c/kuibit/cactus_grid_functions.py#L1443-L1446

The error message is usually 
```py
RuntimeError: File <Closed HDF5 file> cannot be processed
```
This behavior is undesired as a file name should be specified in the error message for debugging. A suggestion is to change the file object to the file name string. (e.g., change `h5f` to `f`)

**Comment by Sbozzolo** [comment]
Thanks. Fixed in 71531df6.

Will be part of 1.4.0

**Comment by Brookluo** [comment]
Thanks for fixing it. The previous fix 
> Implemented in [27c2aff](https://github.com/Sbozzolo/kuibit/commit/27c2aff46d5c46faff153219b95ca863dddead42).

introduces multiple same corrupted file exception handlings. There is inconsistent use of file object and file name/path in the error message f-string. Could you please change all those exceptions referencing file objects to referencing file name strings instead? [71531df](https://github.com/Sbozzolo/kuibit/commit/71531df647e4c974698036903974abd053b7adae) fixes one of the exception handlings.

**Comment by Sbozzolo** [comment]
Thanks!

65b7a3e2 should be address the other ones :)
